torch/_refs/__init__.py

```diff
@@ -3823,7 +3831,7 @@ def _reshape_view_helper(a: TensorLikeType, *shape, allow_copy: bool) -> TensorL
             a_ = flatten(a_, idx, end)
 
         # Splits the (possibly flattened) dimension to create the desired dim length
-        if guard_size_oblivious(accum != length):
+        if guard_or_true(accum == length):
```

Comment:
lets test this one by having a test PR that just remove thos line to verify it does not make a difference.